pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -702,3 +764,55 @@ func (p *staticPolicy) getAlignedCPUs(numaAffinity bitmask.BitMask, allocatableC
 
 	return alignedCPUs
 }
+
+func (p *staticPolicy) validateInPlacePodVerticalScaling(pod *v1.Pod, container *v1.Container) error {
+
+	if v1qos.GetPodQOS(pod) != v1.PodQOSGuaranteed {
+		return nil
+	}
+	cpuQuantity := container.Resources.Requests[v1.ResourceCPU]
+	if cs, ok := podutil.GetContainerStatus(pod.Status.ContainerStatuses, container.Name); ok {
+		allocatedCPUQuantity := cs.AllocatedResources[v1.ResourceCPU]
+		if allocatedCPUQuantity == cpuQuantity {
+			// Skip resize, nothing to be done
+			return inconsistentCPUAllocationError{
```

Comment:
I don't think an error should be return here. If only memory is requested to be resized, this error will prevent the resizing.

Reply:
Thanks for your input, I will update unit tests to include this corner case and update accordingly.

Reply:
Thanks done, i removed it in the latest push.